### PR TITLE
chore: move UITabBarController configuration into new package - WPB-6647

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6647" title="WPB-6647" target="_blank">TopicWPB-6647</a>  [iOS] Navigational Overhaul
  </td></table>
  <br />
 

### Issue

Setting up the tab bar controller is done within a large class (ZClientViewController) but can easily be extracted.
This PR creates a new target `WireUIBase` within the `WireUI` package and moves tab bar controller related code, strings and icons there.

### Testing

Nothing should change from the users' perspective.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [x] Make sure you use the API for UI elements that support large fonts.
- [x] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [x] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: WireUI/Sources/WireUIBase/Resources/en.lproj/Localizable.strings [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+"tabBar.contacts.title" = "Contacts";
```
Comment (caldrian) [inline WireUI/Sources/WireUIBase/Resources/en.lproj/Localizable.strings]: The string has been moved and changed. Can I prevent somehow that the translations aren't lost?

File: .github/workflows/_reusable_run_tests.yml [hunk 2]
```diff
@@ -782,6 +782,10 @@ jobs:
           xcodebuild test -workspace wire-ios-mono.xcworkspace -scheme WireDesign -destination 'platform=iOS Simulator,OS=${{ env.IOS_VERSION }},name=${{ env.IPHONE_MODEL }}' | tee -a xcodebuild-wire-design.log | bundle exec xcpretty --report junit --output build/reports/WireDesign.junit
           exit ${PIPESTATUS[0]}
 
+      # WireUIBase
+
+      # TODO
```
Comment (samwyndham) [inline .github/workflows/_reusable_run_tests.yml]: Should this be done in this PR?

File: WireUI/Sources/WireUIBase/Resources/en.lproj/Localizable.strings (hunk 1 above)
Comment (samwyndham) [reply WireUI/Sources/WireUIBase/Resources/en.lproj/Localizable.strings]: I personally don't know but assume crowdin has a translation memory and would expect that adding the new translations is very fast. Maybe you could forward a list of changed keys to whoever manages crowdin? 🤔

**Review by samwyndham — APPROVED** [review]
Nice work. I didn't check the logic in new swift files such as WireUI/Sources/WireUIBase/MainTabBarController/MainTabBarController.swift` as I assume this code was just moved

File: .github/workflows/_reusable_run_tests.yml (hunk 2 above)
Comment (caldrian) [reply .github/workflows/_reusable_run_tests.yml]: done now

File: WireUI/Sources/WireUIBase/Resources/en.lproj/Localizable.strings (hunk 1 above)
Comment (agisilaos) [reply WireUI/Sources/WireUIBase/Resources/en.lproj/Localizable.strings]: This is an entirely new file even though the same string lives inside. We can check after we merge what's going to trigger the crowdin bot and what are going to be the updates.

File: WireUI/Tests/WireUIBaseTests/MainTabBarController/MainTabBarControllerTabTests.swift
Comment (agisilaos) [inline WireUI/Tests/WireUIBaseTests/MainTabBarController/MainTabBarControllerTabTests.swift:1]: Are you planning to add any unit tests here? @caldrian

File: WireUI/Sources/WireUIBase/MainTabBarController/MainTabBarController.swift [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                    image: .init(systemName: "text.bubble"),
+                    selectedImage: .init(systemName: "text.bubble.fill")
+                )
+                tabBarItem.accessibilityIdentifier = "bottomBarRecentListButton"
+                tabBarItem.accessibilityLabel = NSLocalizedString("tabBar.conversations.description", bundle: .module, comment: "")
+                tabBarItem.accessibilityHint = NSLocalizedString("tabBar.conversations.hint", bundle: .module, comment: "")
+
+            case .folders:
+                tabBarItem = .init(
+                    title: NSLocalizedString("tabBar.folders.title", bundle: .module, comment: ""),
```
Comment (agisilaos) [inline WireUI/Sources/WireUIBase/MainTabBarController/MainTabBarController.swift]: I don't think it's safe setting the title using type `String`. I think we can generate them using `SwiftGen` and use  the `L10n` struct.

**Review by agisilaos — CHANGES_REQUESTED** [review]
Let's integrate `SwiftGen` for generating the `L10n` struct for strings. Using `SwiftGen` to generate the `L10n` struct for localized strings is a much safer and more efficient approach than using raw strings directly.

File: WireUI/Sources/WireUIBase/MainTabBarController/MainTabBarController.swift (hunk 3 above)
Comment (caldrian) [reply WireUI/Sources/WireUIBase/MainTabBarController/MainTabBarController.swift]: Using the initialiser for string catalog now.

File: WireUI/Tests/WireUIBaseTests/MainTabBarController/MainTabBarControllerTabTests.swift
Comment (caldrian) [reply WireUI/Tests/WireUIBaseTests/MainTabBarController/MainTabBarControllerTabTests.swift:1]: Done, snapshot tests.

File: WireUI/Sources/WireUIBase/Resources/en.lproj/Localizable.strings (hunk 1 above)
Comment (caldrian) [reply WireUI/Sources/WireUIBase/Resources/en.lproj/Localizable.strings]: Solved it by running the Crowdin action after creating the new files, before deleting the old ones. Seems like Crowdin is then able to fill in the duplicate entries properly.

**Review by agisilaos — APPROVED** [review]

**Pull request merged**